tests/test_utils.py

```diff
@@ -0,0 +1,360 @@
+        )
+
+        assert result == 'Fedora-Atomic-26-1.5.x86_64.raw.xz'
+
+    @mock.patch('fedimg.utils.external_run_command')
+    def test_get_source_from_image(self, mock_erc):
+        mock_erc.return_value = (
+            "2018-03-27 00:00:00 (93 KB/s) 'Fedora-Atomic-26-1.5.x86_64.raw.xz' (1234569/123456789)",
+            "",
+            0
+        )
+        fedimg.utils.get_source_from_image('https://somepage.org/Fedora-Atomic-26-1.5.x86_64.raw.xz')
+
+        mock_erc.assert_called_once_with([
+            'wget',
+            'https://somepage.org/Fedora-Atomic-26-1.5.x86_64.raw.xz',
+            '-P',
+            mock.ANY
+        ])
+
+    @mock.patch('fedimg.utils.external_run_command')
+    def test_get_source_from_image_retcode_1(self, mock_erc):
+        mock_erc.return_value = (
+            "2018-03-27 00:00:00 (93 KB/s) 'Fedora-Atomic-26-1.5.x86_64.raw.xz' (1234569/123456789)",
+            "",
+            1
+        )
+        file_path = fedimg.utils.get_source_from_image('https://somepage.org/Fedora-Atomic-26-1.5.x86_64.raw.xz')
+
+        mock_erc.assert_called_once_with([
+            'wget',
+            'https://somepage.org/Fedora-Atomic-26-1.5.x86_64.raw.xz',
+            '-P',
+            mock.ANY
+        ])
+        assert file_path == ''
+
+    def test_get_volume_type_from_image(self):
+        mock_image = mock.Mock()
+        mock_image.extra = {
+            'block_device_mapping': [{
+                'ebs': {
+                    'volume_type': 'gp2'
+                }
+            }]
+        }
+        region = 'us-east-1'
+
+        volume_type = fedimg.utils.get_volume_type_from_image(
+                mock_image, region)
+
+        assert volume_type == 'gp2'
+
+    def test_get_virt_type_from_image_hvm(self):
+        mock_image = mock.Mock()
+        mock_image.extra = {
+            'block_device_mapping': [{
+                'device_name': '/dev/sda1'
+            }]
+        }
+
+        virt_type = fedimg.utils.get_virt_type_from_image(mock_image)
+
+        assert virt_type == 'hvm'
+
+    def test_get_virt_type_from_image_paravirtual(self):
+        mock_image = mock.Mock()
+        mock_image.extra = {
+            'block_device_mapping': [{
+                'device_name': '/dev/sda'
+            }]
+        }
+
+        virt_type = fedimg.utils.get_virt_type_from_image(mock_image)
+
+        assert virt_type == 'paravirtual'
+
+    def test_region_to_driver(self):
+        driver_1 = fedimg.utils.region_to_driver('us-east-1')
+        driver_2 = fedimg.utils.region_to_driver('eu-west-1')
+
+        assert driver_1.func == driver_2.func
+
+    @mock.patch('paramiko.SSHClient')
+    def test_ssh_connection_works_true(self, mocksshclient):
+        mocksshclient.connect.return_value = ''
+        result = fedimg.utils.ssh_connection_works(
+            'testuser',
+            '127.0.0.1',
+            '/path/to/key'
+        )
+
+        assert result == True
```

Comment:
If ```result``` is supposed to be a ```bool```, the above assertion actually does not check that because ```1 == True``` is ```True``` too. This will assert that ```result``` is a bool:

```
self.assertIs(result, True)
```